org.jenerate/src/test/org/jenerate/internal/strategy/method/content/impl/java/JavaEqualsMethodContentTest.java

```diff
@@ -159,4 +160,30 @@ public void testGetMethodContentWithClassCastAndUseBlocksInIfStatements() throws
                         + "Objects.equals(field1, castOther.field1) && Objects.equals(field2, castOther.field2);\n",
                 content);
     }
+
+    @Test
+    public void testGetMethodContentWithClassComparisonAndUseSimplePrimitiveComparison() throws Exception {
+        when(data.useClassComparison()).thenReturn(true);
+        when(data.useBlockInIfStatements()).thenReturn(true);
+        when(data.useSimplePrimitiveComparison()).thenReturn(true);
+        String content = methodContent.getMethodContent(objectClass, data);
+        assertEquals(
+                "if (other == null){\n return false;\n}\nif ( !getClass().equals(other.getClass()))"
+                        + "{\n return false;\n}\nTest castOther = (Test) other;\nreturn "
+                        + "(field1 == castOther.field1) && Objects.equal(field2, castOther.field2);\n",
+                content);
+    }
+
+    @Test
+    public void testGetMethodContentWithClassComparisonAndUseDeepArrayComparison() throws Exception {
+        when(data.useClassComparison()).thenReturn(true);
+        when(data.useBlockInIfStatements()).thenReturn(true);
+        when(data.useDeepArrayComparison()).thenReturn(true);
+        String content = methodContent.getMethodContent(objectClass, data);
+        assertEquals(
+                "if (other == null){\n return false;\n}\nif ( !getClass().equals(other.getClass()))"
+                        + "{\n return false;\n}\nTest castOther = (Test) other;\nreturn "
+                        + "Objects.equal(field1, castOther.field1) && Arrays.equals(field2, castOther.field2);\n",

```

Comment:
Isn't it Objects.equal**s**(field1, castOther.field1)?

Reply:
Definitely it should be equals